### PR TITLE
feat(ecs-patterns): allow specifying security groups on ScheduledTask pattern

### LINKED ISSUE #5213: (ecs-patterns): use existing Security Group with ScheduledFargateTask
## :question: General Issue
Unable to find a way to specify an existing security group when creating task schedules for fargate tasks using aws cdk

### The Question

When defining an ECS Task Schedule, I can't seem to find a way of specifying an existing security group. Any pointers on where this can be configured using aws cdk?

In the code snippet below, you'll see I am able to create a cron, specify the docker image to schedule and create the schedule itself by specifying the existing cluster and vpc. However, there is no option to specify an existing security group... Is it possible to specify an existing security group?

``` python
schedule_cron = scaling.Schedule.cron(minute=manifest['schedule']['minute'], 
                                                hour=manifest['schedule']['hour'], 
                                                day=manifest['schedule']['day'], 
                                                month=manifest['schedule']['month'], 
                                                year=manifest['schedule']['year'])

image_option = ecs_patterns.ScheduledFargateTaskImageOptions(image=img, 
                                                cpu=manifest["resources"]["cpu"], 
                                                memory_limit_mib=manifest["resources"]["memory"],
                                                log_driver=ecs.AwsLogDriver(log_group=log_group,
                                                                            stream_prefix=manifest["app_name"]),
                                                secrets=secrets,
                                                environment= env)

schedule_pattern = ecs_patterns.ScheduledFargateTask(self, f"scheduledtask{app_name}", 
                                                schedule= schedule_cron, scheduled_fargate_task_image_options=image_option, cluster=cluster, 
                                                desired_task_count=manifest["replica_count"], vpc=vpc)

```

When i do a `cdk synth` i can see that cdk is trying to generate a new security group with some outbound and inbound rules. Instead, i would like to pass an existing security group that's been defined with relevant organization access policies. Is it supported? 

### Environment

  - **CDK CLI Version:** 1.15.0
  - **Module Version:** 1.15.0
  - **OS:** Windows
  - **Language:** Python

### Other information

https://stackoverflow.com/questions/59067514/aws-cdk-ecs-task-scheduling-specify-existing-securitygroup

### PR DESCRIPTION
**Pull request by SoManyHs**
Closes #5213, #14220

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-ecs-patterns/lib/base/scheduled-task-base.ts
Comment (madeline-k) [inline packages/@aws-cdk/aws-ecs-patterns/lib/base/scheduled-task-base.ts:165]: What if the user does not provide this field? The doc comment says the a new security group will be created. At what point will this.securityGroups be assigned to the newly created default security group?

Comment (SoManyHs) [reply packages/@aws-cdk/aws-ecs-patterns/lib/base/scheduled-task-base.ts:165]: Good question -- it's not totally obvious, but it gets created further down the stack within the events-target lib: https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-events-targets/lib/ecs-task.ts#L147-L150

Comment (madeline-k) [reply packages/@aws-cdk/aws-ecs-patterns/lib/base/scheduled-task-base.ts:165]: Okay thanks! I see now where the security group gets created. However, I did a quick test and after creating the security group, it is never assigned to the public securityGroups field of the ECS Task. Basically in the case where the customer omits the securityGroups prop:

```
const scheduledFargateTask = new ScheduledFargateTask(stack, 'ScheduledFargateTask', { 
  cluster, 
  scheduledFargateTaskImageOptions: {    
    image: ecs.ContainerImage.fromRegistry('amazon/amazon-ecs-sample'),    
    memoryLimitMiB: 512,
  },  
  schedule: events.Schedule.expression('rate(1 minute)'),  
});
```

then `scheduledFargateTask.securityGroups` will be undefined. Is that the intended behavior? It seems to me that this newly introduced securityGroups field should still give the user the ability to access the associated security group even if they didn't pass in a custom one.

Comment (madeline-k) [reply packages/@aws-cdk/aws-ecs-patterns/lib/base/scheduled-task-base.ts:165]: I noticed this PR will make the securityGroups available by surfacing the entire EcsTask: https://github.com/aws/aws-cdk/pull/14953/files. So I think we can remove this output property in ScheduledTaskBase.

**Review by madeline-k — CHANGES_REQUESTED** [review]
Looks good! Will approve once you remove the public `ScheduledTaskBase.securityGroups` field.

File: packages/@aws-cdk/aws-ecs-patterns/lib/base/scheduled-task-base.ts
Comment (madeline-k) [inline packages/@aws-cdk/aws-ecs-patterns/lib/base/scheduled-task-base.ts:151]: nit: Add newline before, remove newline after.
```suggestion
  
  /**
   * The security group to use for the ECS Task.
   */
  private readonly _securityGroups?: ISecurityGroup[];
```

**Review by madeline-k — APPROVED** [review]

**Comment by madeline-k** [comment]
Added do-not-merge label so that it won't merge before you decide whether or not to take my suggestion. @SoManyHs remove the label when you're ready.

**Pull request merged**
